share/functions/cd.fish

```diff
@@ -3,10 +3,17 @@
 #
 function cd --description "Change directory"
     set -l MAX_DIR_HIST 25
+    set -l options h/help L/logical P/physical
+    argparse -n cd --exclusive L,P --max-args=1 $options -- $argv
+    or return
 
-    if test (count $argv) -gt (test "$argv[1]" = "--" && echo 2 || echo 1)
-        printf "%s\n" (_ "Too many args for cd command")
-        return 1
+    # The argparse removes the options from argv. When the option -L is uses
+    # it's not necessary to do something as fish always resolves the directory
+    # Like POSIX it's the default option for cd.
+
+    # Drop the argument and use the realpath
+    if set -q _flag_P
+        set argv (realpath $argv)
```

Comment:
There are some subtle problems here:

1. We have a realpath *builtin*, but we special case it because some realpaths do more, so it's only used with `builtin realpath`. Using the builtin here is preferable for performance and consistency reasons.

2. `realpath` prints its own error if more than one component of the path doesn't exist (so "foo" always works, but "foo/bar" doesn't if foo doesn't exist). This would print both the realpath error and the cd error. You should probably silence realpath and only set argv if it succeeds. 

3. If you used `cd -P -- -foo` to go to a directory called `-foo`, this would try to interpret it as an option - it should be `realpath -- $argv` (or rather `builtin realpath -- $argv`)